### ISSUE TITLE
Prevent fat-fingered double submission of registration

### ISSUE THREAD

**Issue by stevepiercy** [bikeindex/bike_index#447]
I got fat fingers when registering bikes, and found that 2 of the 20 bikes I registered at Sunday Streets under GEARs yesterday were duplicates.

We should implement a feature to prevent double submission. I would suggest a JavaScript that disables the submit button when it is tapped or clicked.

**Comment by sethherr** [comment]
This is definitely an important feature. 

We’re you registering them under the organizations page or using the normal flow?

**Comment by stevepiercy** [comment]
Under the [GEARs organization](https://eugenegears.org/bike-registration/), using my Android phone, Chrome browser.

**Comment by sethherr** [comment]
Ok good. It should be prevented for the normal flow.

I'll work on the JS unless you want to add it yourself @stevepiercy

**Comment by stevepiercy** [comment]
It looks like the form will send a POST request on submit and does not use any XHR, correct?

And it seems that you're using jQuery via Twitter Bootstrap, but I'm not 100% certain that it gets compiled into the assets.

Also one should consider more than just clicking the submit button. This SO answer to the problem has some details and uses jQuery.

https://stackoverflow.com/a/12952154/2214933

**Comment by sethherr** [comment]
Yup, no XHR there. And jQuery does get into the assets.

That SO answer looks like a great solution.

If you're comfortable with coffeescript (I know, I know, ES6 is great and I want to move us there soon), this is the script file [/app/assets/javascripts/external_scripts/embed_script.js.coffee](https://github.com/bikeindex/bike_index/blob/master/app/assets/javascripts/external_scripts/embed_script.js.coffee), otherwise let me know and I'll add it 🙂

**Comment by stevepiercy** [comment]
See PR #449.

1. CS is foreign to me, so my PR #449 is probably wrong. Here's what I tried to adapt.
    ```javascript
    $('#new_bike').submit(function(){
        $this = $(this);
        /** prevent double posting */
        if ($this.data().isSubmitted) {
            return false;
        }
        /** do some processing */
        /** mark the form as processed, so we will not process it again */
        $this.data().isSubmitted = true;
        return true;
    });
    ```
1. As an aside, I like me some [webpack](https://webpack.js.org/) for bundling. Let me know if you need help with that.
1. Do you have a branch for work on switching to ES6?
1. Per [this note](https://github.com/bikeindex/bike_index/blob/master/app/assets/javascripts/external_scripts/embed_script.js.coffee#L44), do you mean pay for hosting TLS on Heroku or purchasing a cert? If the latter, consider Let's Encrypt.

**Comment by sethherr** [comment]
1. I commented on the PR
2. Sweet. I was planning to upgrade to Rails 5 first, since it includes [webpacker](https://github.com/rails/webpacker) - but looks like webpacker supports 4.2
3. There isn't a branch for adding webpacker yet. I have _some_ experience adding webpacker to rails apps. I'll create an issue for it. After we have webpacker added, we can migrate bits of js over - I'll let you know when that happens! There are specific coffee > ES6 converters available
4. Pay for hosting TLS on Heroku. We're using Let's Encrypt in at least one place, it's awesome!

**Comment by stevepiercy** [comment]
2. Note that the `stable` branch supports webpack 3.x, whereas `master` supports 4.x. I'd suggest the latter to avoid two upgrades.
3. FYI, my proficiency in Rails is novice, whereas I am competent in Python and just enough JS to survive. Speaking of which, I highly recommend [SurviveJS](https://survivejs.com/) as an excellent resource to learn webpack. Fun fact, I helped the author by editing the first version of his book, which forced me to learn and retain the core concepts. There have been lots of changes, but the core remains.
4. Not knowing your hosting needs, but I've noticed occasional outages during the wee hours of the morning (3am Pacific), and I wonder if it's due to the dyno going to sleep after 30 minutes of activity? A reload of the page "fixes" the issue. Is a change to hosting under consideration?

**Comment by sethherr** [comment]
2. Sounds good - I’ll make sure to get on 4.x when adding it. It’s probably going to be a little while before I add it. 
3. I’ll take a look at that book!
4. Bike Index is hosted on [packet.net](https://packet.net) servers and deployed with [cloud66](https://cloud66) right now. I think it might be time to up the size of our primary server, thanks for saying something about late night timeouts!

I’m closing this issue now, since it was fixed by #449
